backend/functions/lib/adapters/gateways/mongo/orders.mongo.repository.ts

```diff
@@ -16,10 +16,33 @@ const COLLECTION_NAME = 'orders';
 const { contains, equals, orderBy } = QueryBuilder;
 
 export class OrdersMongoRepository extends BaseMongoRepository implements OrdersRepository {
-  constructor(context: ApplicationContext) {
+  private static referenceCount: number = 0;
+  private static instance: OrdersMongoRepository;
+
+  private constructor(context: ApplicationContext) {
     super(context, MODULE_NAME, COLLECTION_NAME);
   }
 
+  public static getInstance(context: ApplicationContext) {
+    if (!OrdersMongoRepository.instance)
+      OrdersMongoRepository.instance = new OrdersMongoRepository(context);
+    OrdersMongoRepository.referenceCount++;
+    return OrdersMongoRepository.instance;
+  }
+
+  public static dropInstance() {
+    if (OrdersMongoRepository.referenceCount > 0) OrdersMongoRepository.referenceCount--;
+    if (OrdersMongoRepository.referenceCount < 1) {
+      OrdersMongoRepository.instance.client.close().then();
```

Comment:
Never tried this out but what happens when an error is thrown and we don't have an error handler in the chain?